lib/handlers/run.js

```diff
@@ -126,9 +142,39 @@ function run(req, res, next, options) {
                         if (errOrStop === false) {
                             return next(false);
                         }
-                        // no need to check for case #2, an error object will
-                        // get propagated all the way up to restify through
-                        // the nested vasyncs.
+
+                        // If this is set to something other than undefined
+                        // we should pass the real error object on and ignore
+                        // the sharding
+                        if (errOrStop) {
+                            return handlerCb(errOrStop);
+                        }
+
+                        // if the conductor we started with is not equal to the current
+                        // conductor, we sharded! in this scenario, we want to bail out
+                        // of the current vasync process, and restart it again with the
+                        // new conductor, at the next numerical key.
+                        var _currentConductor = reqHelpers.getConductor(req);
```

Comment:
If I'm reading this correctly, are you checking for sharding within the middle of a handler block? e.g., [a, b, c] and we sharded at a, so we want to skip b and c. 

If that's the case, then we probably don't need to check for sharding at the beginning of each block, on line 79?